src/ansys/pytwin/evaluate/tbrom.py

```diff
@@ -178,31 +176,19 @@ def _read_basis(filepath):
             var = struct.unpack("cccccccccccccccc", f.read(16))[0]
             nb_val = struct.unpack("Q", f.read(8))[0]
             nb_mc = struct.unpack("Q", f.read(8))[0]
-            basis = []
-            for i in range(nb_mc):
-                vec = []
-                for j in range(nb_val):
-                    vec.append(struct.unpack("d", f.read(8))[0])
-                basis.append(vec)
-        return basis
+        return np.fromfile(filepath, dtype=np.double, offset=32).reshape(-1, nb_val)
```

Comment:
Not 100% sure how `np.fromfile` works (great find, btw), but it looks like we could also pass is an open file object, so maybe use
```python
with open(filepath, "rb") as f:
    ...
    return np.fromfile(f, dtype=np.double, offset=0).reshape(-1, nb_val)
```
We could then avoid opening the file twice.